src/ensembl/src/content/app/entity-viewer/gene-view/GeneView.tsx

```diff
@@ -153,30 +147,58 @@ const GeneViewWithData = (props: GeneViewWithDataProps) => {
         <GeneViewTabs />
       </div>
       <div className={styles.geneViewTabContent}>
-        {props.selectedGeneTabName === GeneViewTabName.TRANSCRIPTS &&
+        {selectedTabs.primaryTab === GeneViewTabName.TRANSCRIPTS &&
           basePairsRulerTicks && (
             <DefaultTranscriptslist
               gene={props.gene}
               rulerTicks={basePairsRulerTicks}
             />
           )}
 
-        {props.selectedGeneTabName === GeneViewTabName.GENE_FUNCTION && (
+        {selectedTabs.primaryTab === GeneViewTabName.GENE_FUNCTION && (
           <GeneFunction gene={props.gene} />
         )}
 
-        {props.selectedGeneTabName === GeneViewTabName.GENE_RELATIONSHIPS && (
+        {selectedTabs.primaryTab === GeneViewTabName.GENE_RELATIONSHIPS && (
           <GeneRelationships />
         )}
       </div>
     </div>
   );
 };
 
-const mapStateToProps = (state: RootState) => ({
-  // FIXME: this will have to be superseded with a proper way we get ids
-  geneId: getEntityViewerActiveEnsObject(state)?.stable_id || null,
-  selectedGeneTabName: getEntityViewerActiveGeneTab(state)
-});
+const useGeneViewRouting = () => {
+  const dispatch = useDispatch();
+  const params: { [key: string]: string } = useParams();
+  const { genomeId, entityId } = params;
+  const { objectId: geneId } = parseFocusIdFromUrl(entityId);
+  const { search } = useLocation();
+  // TODO: discuss – is using URLSearchParams better than using the querystring package?
+  const view = new URLSearchParams(search).get('view');
```

Comment:
This looks simple and easy to me.

Reply:
We should review whether we want to continue using the `query-string` module then. The only significant differences seem to be that the `query-string` module is more flexible with the handling of array query parameters, and it also handles empty values nicer. On the one hand, URLSearchParams is native web api; but on the other, `query-string` is a fairly small library, 5kB minified.

Reply:
We could probably continue to use `URLSearchParams` and then go for `query-string` when we come across some complex queries involving arrays and empty values.

Reply:
I vaguely remember us having similar discussions in one of the PRs where we decided to go for `query-string`. I think we can leave it as it is for now and try to use `URLSearchParams` wherever we can and use `query-string` to handle complex queries.

Reply:
I guess our thinking should be:
- if we want to use query-string anyway, we probably should use it consistently throughout
- if URLSearchParams meets our needs and we don't really need query-string, we should get rid of it and of those extra 5kB of minified javascript that it contributes to our js bundle

Reply:
I noticed that it is currently used in the following places:
https://github.com/Ensembl/ensembl-client/blob/dev/src/ensembl/src/content/app/species-selector/state/speciesSelectorEpics.ts#L66
https://github.com/Ensembl/ensembl-client/blob/dev/src/ensembl/src/shared/helpers/urlHelper.ts

@azangru since you have worked on these files, if you think it can be easily replaced with `URLSearchParams`, I guess we can get rid of it in a separate PR.